python/mspasspy/global_history/manager.py

```diff
@@ -1,21 +1,173 @@
 import os
 import yaml
 import pymongo
-import pyspark
 import collections
-import dask.bag as daskbag
 import json
 
+try:
+    import dask.bag as daskbag
+    import pyspark
+except:
+    pass
 from bson.objectid import ObjectId
 from mspasspy.ccore.utility import MsPASSError, AntelopePf
 from mspasspy.util.converter import AntelopePf2dict
 from datetime import datetime
 from dill.source import getsource
+import functools
 
 import mspasspy.algorithms.signals as signals
 from mspasspy.global_history.ParameterGTree import ParameterGTree, parameter_to_GTree
 
 
+def mspass_normal_map(
```

Comment:
Please name them as `mspass_map` and `mspass_reduce` for consistency. Note that those are the same name we insert into the Bag and RDD.